File: frontend/src/features/sabor_familia/service/PerfilService.ts
import api from "../lib/Api";
import { criarFormDataMultipart } from "../lib/multipartUtils";
import type { PerfilResponse } from "../dto/perfil/response/PerfilResponse";
import type { PerfilRequest } from "../dto/perfil/request/PerfilRequest";
import type { EditarPerfilRequest } from "../dto/perfil/request/EditarPerfilRequest";
import type { PerfilResumoResponse } from "../dto/perfil/response/PerfilResumoResponse";
import type { PageResponse } from "../dto/page/PageResponse";
import type { SeguindoResponse } from "../dto/seguindo/SeguindoResponse";

export const perfilService = {
  buscarPerfilPublico: async (perfilId: number): Promise<PerfilResponse> => {
    const { data } = await api.get<PerfilResponse>(`/perfil/${perfilId}`);
    return data;
  },

  buscarMeuPerfil: async (): Promise<PerfilResponse> => {
    const { data } = await api.get<PerfilResponse>("/perfil");
    return data;
  },

  explorarPerfis: async (
    page = 0,
    size = 20,
    nome?: string
  ): Promise<PageResponse<PerfilResumoResponse>> => {
    const { data } = await api.get<PageResponse<PerfilResumoResponse>>("/perfil/explorar", {
      params: {
        page,
        size,
        sort: "dataCadastro,desc",
        ...(nome && { nome }),
      },
    });
    return data;
  },

  criarPerfil: async (
    request: PerfilRequest,
    arquivo?: File | null
  ): Promise<PerfilResponse> => {
    const formData = criarFormDataMultipart(request, arquivo);
    const { data } = await api.post<PerfilResponse>("/perfil", formData);
    return data;
  },
 
  editarPerfil: async (
    request: EditarPerfilRequest,
    arquivo?: File | null
  ): Promise<PerfilResponse> => {
    const formData = criarFormDataMultipart(request, arquivo);
    const { data } = await api.put<PerfilResponse>("/perfil", formData);
    return data;
  },
 
  buscarSeguidores: async (
    perfilId: number,
    page = 0,
    size = 20
  ): Promise<PageResponse<PerfilResumoResponse>> => {
    const { data } = await api.get<PageResponse<PerfilResumoResponse>>(
      `/perfil/${perfilId}/seguidores`,
      { params: { page, size } }
    );
    return data;
  },
 
  buscarSeguindo: async (
    perfilId: number,
    page = 0,
    size = 20
  ): Promise<PageResponse<PerfilResumoResponse>> => {
    const { data } = await api.get<PageResponse<PerfilResumoResponse>>(
      `/perfil/${perfilId}/seguindo`,
      { params: { page, size } }
    );
    return data;
  },
 
  seguirPerfil: async (perfilId: number): Promise<SeguindoResponse> => {
    const { data } = await api.post<SeguindoResponse>(`/perfil/${perfilId}/seguir`);
    return data;
  },
 
  deixarSeguirPerfil: async (perfilId: number): Promise<SeguindoResponse> => {
    const { data } = await api.delete<SeguindoResponse>(`/perfil/${perfilId}/seguir`);
    return data;
  },

};

File: backend/modules/sabor-familia/src/main/java/br/com/puc/saborfamilia/service/perfil/PerfilService.java
package br.com.puc.saborfamilia.service.perfil;

import br.com.puc.saborfamilia.service.perfil.dto.request.EditarPerfilRequest;
import br.com.puc.saborfamilia.service.perfil.dto.request.PerfilRequest;
import br.com.puc.saborfamilia.service.perfil.dto.response.PerfilResumoResponse;
import br.com.puc.saborfamilia.service.perfil.dto.response.PerfilResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

public interface PerfilService {

  PerfilResponse buscarMeuPerfil(Long usuarioId);

  PerfilResponse buscarPerfilPublico(Long usuarioId, Long perfilId);

  Page<PerfilResumoResponse> explorarPerfis(Long usuarioId, String nome, Pageable pageable);

  PerfilResponse criarPerfil(Long usuarioId, PerfilRequest request, MultipartFile fotoPerfil);

  PerfilResponse editarPerfil(Long usuarioId, EditarPerfilRequest request, MultipartFile fotoPerfil);

}
